### ISSUE TITLE
`rrelu` gives wrong gradient on cuda for particular inputs

### ISSUE THREAD

**Issue by upupyy** [pytorch/pytorch#80205]
### 🐛 Describe the bug

`rrelu` gives wrong gradient on cuda for particular inputs

```python
import torch

def get_fn():
    lower = -2.9
    upper = -2.7
    training = True
    def fn(input):
        fn_res = torch.nn.functional.rrelu(input, lower=lower, upper=upper, training=training)
        return fn_res
    return fn

input_tensor = torch.tensor([0.1250, 0.4313], dtype=torch.float64)
fn = get_fn()
try:
    input = input_tensor.clone().detach().to('cuda').requires_grad_()
    torch.autograd.gradcheck(fn, (input), check_sparse_nnz=False, atol=0.01, rtol=0.01, check_forward_ad=False, check_backward_ad=True, check_batched_grad=False)
except Exception as e:
    print(e)

# Jacobian mismatch for output 0 with respect to input 0,
# numerical:tensor([[1.0000, 0.0000],
#         [0.0000, 1.0000]], device='cuda:0', dtype=torch.float64)
# analytical:tensor([[0., 0.],
#         [0., 0.]], device='cuda:0', dtype=torch.float64)
```

Obviously, the gradient shouldn't be 0 since the input tensor is greater than 0. This cannot reproduce on cpu.

### Versions

pytorch: 1.11.0
cuda: 11.3
RTX 6000

cc @ezyang @gchanan @zou3519 @albanD @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are @ngimel

**Comment by ngimel** [comment]
It seems like something is going wrong with the gradcheck, because usual `.backward` produces correct gradients
